examples/kaggle_pipeline.py

```diff
@@ -0,0 +1,292 @@
+        }
+
+        for name, method in methods.items():
+            logger.info(f"\nEvaluating {name}...")
+            try:
+                X_selected = method(X_train.copy(), y_train)
+                if self.tune_hyperparameters and hasattr(self.model, "fit"):
+                    logger.info("Performing hyperparameter tuning...")
+                    self.model.fit(X_selected, y_train)
+                    logger.info("Best parameters found:")
+                    logger.info(self.model.best_params_)
+                    logger.info(f"Best cross-validation MAE: {-self.model.best_score_:.4f}")
+
+                scores = self._get_cv_scores(X_selected, y_train, cv)
+                results[name] = {
+                    "n_features": X_selected.shape[1],
+                    "mean_mae": np.mean(scores),
+                    "std_mae": np.std(scores),
+                    "selected_features": list(X_selected.columns),
+                }
+                logger.info(f"Number of features: {X_selected.shape[1]}")
+                logger.info(f"Mean MAE: {np.mean(scores):.4f} (+/- {np.std(scores) * 2:.4f})")
+            except Exception as e:
+                logger.error(f"Error with {name}: {e!s}")
+
+        return results
+
+    def _get_cv_scores(self, X, y, cv):
+        """Helper function to get cross-validation scores"""
+        scores = cross_val_score(
+            self.model,
+            X,
+            y,
+            cv=KFold(n_splits=cv, shuffle=True, random_state=42),
+            scoring="neg_mean_absolute_error",
+            n_jobs=-1,
+        )
+        return -scores
+
+    def select_features(self, X_train, y_train, X_test, method="model_based"):
+        """Apply selected feature selection method to both train and test"""
+        if method == "mutual_info":
+            X_train_selected = self.feature_selector.mutual_information(X_train, y_train)
+        elif method == "model_based":
+            X_train_selected = self.feature_selector.model_based_selection(X_train, y_train)
+        elif method == "boruta":
+            X_train_selected = self.feature_selector.boruta_selection(X_train, y_train)
+        else:
+            return X_train, X_test
+
+        selected_features = X_train_selected.columns
+        X_test_selected = X_test[selected_features]
+        return X_train_selected, X_test_selected
+
+    def validate_model(self, X_train, y_train, cv=10):
+        """Perform cross-validation and print results"""
+        scores = self._get_cv_scores(X_train, y_train, cv)
+        self.cv_scores = scores
+        logger.info(f"Cross-validation MAE scores: {scores}")
+        logger.info(f"Mean MAE: {np.mean(scores):.4f} (+/- {np.std(scores) * 2:.4f})")
+
+    def train_and_predict(self, X_train, y_train, X_test):
+        """Train model and generate predictions"""
+        logger.info("Training final model...")
+
+        if self.tune_hyperparameters and hasattr(self.model, "best_estimator_"):
```

Comment:
having if/else here seems a bit messy and it can be a source of headache.. better have the best model always places in the same attribute (whether passed from the init or tuned) and then just run with it.